src/symbolicConstants.py

```diff
@@ -0,0 +1,193 @@
+and makes this module aware of the cache by adding the _addSymConstToTable
+function to this module.
+Outside Abaqus/CAE, we do not have the same requirement.
+"""
+
+from __future__ import annotations
+from typing import Any
+
+# imports
+import re
+import sys
+import os
+
+# classes
+
+class AbaqusBoolean(int):
+    """
+    AbaqusBoolean is derived from int, and prints as ON or OFF.
+    A call to SymbolicConstant("ON/OFF") will return an AbaqusBoolean instance.
+    """
+
+    def getId(self) -> int:
+        ...
+
+    def getText(self) -> str:
+        if bool(self):
+            return 'ON'
+        else:
+            return 'OFF'
+
+    def isTrue(self) -> bool:
+        return bool(self)
+
+    def __copy__(self) -> AbaqusBoolean:
+        ...
+    
+    def __getstate__(self) -> bool:
+        ...
+    
+    def __hash__(self) -> int:
+        ...
+
+    def __init__(self, value:int) -> None:
+        super().__init__()
+        if value not in (0, 1):
+            raise ValueError(f'AbaqusBoolean must have value argument 0 or 1.  {value} supplied')
+
+    @staticmethod  # known case of __new__
+    def __new__(cls, *args, **kargs):  # reliably restored by inspect
+        # no doc
+        pass
+
+    def __reduce__(self) -> tuple:  # known return case of __reduce__
+        ...
+    
+    def __repr__(self) -> str: 
+        return self.getText()
+
+    def __setstate__(self, *args) -> None:
+        ...
+    
+    def __str__(self) -> str:
+        return self.getText()
+
+    __weakref__ = property(lambda self: object(), lambda self, v: None, lambda self: None)  # default
+    """list of weak references to the object (if defined)"""
+
+    _abaqusBooleanOFF = 0
+    _abaqusBooleanON = 1
+    __dict__ = None  # (!) real value is 'dict_proxy({\'__module__\': \'symbolicConstants\', \'isTrue\': <function isTrue at 0x00000000031652E8>, \'__setstate__\': <function __setstate__ at 0x00000000031653C8>, \'__str__\': <function getText at 0x0000000003165208>, \'__reduce__\': <function __reduce__ at 0x00000000031654A8>, \'getText\': <function getText at 0x0000000003165208>, \'__dict__\': <attribute \'__dict__\' of \'AbaqusBoolean\' objects>, \'_abaqusBooleanON\': ON, \'__weakref__\': <attribute \'__weakref__\' of \'AbaqusBoolean\' objects>, \'_abaqusBooleanOFF\': OFF, \'__copy__\': <function __copy__ at 0x0000000003165438>, \'__new__\': <staticmethod object at 0x000000000314E6A8>, \'getId\': <function getId at 0x0000000003165278>, \'__init__\': <function __init__ at 0x0000000003165128>, \'__hash__\': <function __hash__ at 0x0000000003165198>, \'__getstate__\': <function __getstate__ at 0x0000000003165358>, \'__doc__\': \'\\n    AbaqusBoolean is derived from int, and prints as ON or OFF.\\n    A call to SymbolicConstant("ON/OFF") will return an AbaqusBoolean instance.\\n    \', \'__repr__\': <function getText at 0x0000000003165208>})'
+
+
+AbaqusBooleanType = AbaqusBoolean
+
+BooleanType = bool
+
+Boolean = AbaqusBoolean
```

Comment:
@haiiliin
I've decided to do this modification instead of modify all type hints of `Boolean` to `AbaqusBoolean` in the other files. Please, see if it is acceptable.

Reply:
I think it is good